src/overlay005/field_control.c

```diff
@@ -174,22 +174,23 @@ void FieldInput_Update (FieldInput *input, FieldSystem *fieldSystem, u16 pressed
     input->playerDir = sub_02061308(fieldSystem->playerAvatar, pressedKeys, heldKeys);
 }
 
-int FieldInput_Process (const FieldInput *input, FieldSystem *fieldSystem)
+BOOL FieldInput_Process (const FieldInput *input, FieldSystem *fieldSystem)
 {
-    if (input->unused5 == FALSE && sub_0203F5C0(fieldSystem, 1) == TRUE) {
+    if (input->dummy5 == FALSE && sub_0203F5C0(fieldSystem, 1) == TRUE) {
         return TRUE;
     }
 
-    if (input->unused5 == FALSE) {
-        BOOL doublesCheck = sub_02054AB0(Party_GetFromSavedata(fieldSystem->saveData));
+    // dummy5 will always be false, so this branch will always be taken, but it doesn't match without the condition
+    if (input->dummy5 == FALSE) {
+        BOOL hasTwoAliveMons = Party_HasTwoAliveMons(Party_GetFromSavedata(fieldSystem->saveData));
 
         if (sub_0206A984(SaveData_Events(fieldSystem->saveData)) == TRUE) {
-            doublesCheck = TRUE;
+            hasTwoAliveMons = TRUE;
         }
 
-        if (sub_02067A84(fieldSystem, doublesCheck) == TRUE || 
-            sub_02071CB4(fieldSystem, 2) == TRUE && 
-            ov8_0224C51C(fieldSystem) == TRUE) {
+        if (sub_02067A84(fieldSystem, hasTwoAliveMons) == TRUE
+            || sub_02071CB4(fieldSystem, 2) == TRUE
+            && ov8_0224C51C(fieldSystem) == TRUE) {
```

Comment:
Is it worth wrapping the conjunction here into a set of parens, to clarify the order-of-operations?